iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/LocationShifter.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.kubernetes.visitors;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.sonar.iac.common.api.tree.impl.TextPointer;
+import org.sonar.iac.common.api.tree.impl.TextRange;
+
+/**
+ * This class is used to store all lines that has to be shifted.
+ * The data are stored into this class through methods {@link #addLineSize(int)} and {@link #addShiftedLine(int, int)}.
+ * Then we can use those data through the method {@link #computeShiftedLocation(TextRange)}, which for a given {@link TextRange} will provide
+ * a shifted {@link TextRange}.
+ * This is especially used in helm context, when the issue we are detecting on the transformed code should be raised on the original code.
+ */
+public class LocationShifter {
```

Comment:
I don't think the Shifter will work on multiple files in a row. IMO, this should be tested more thoroughly.

Reply:
An alternative could also be to do as you suggested: contextualize the data with the `InputFileContext`.
We would have a data structure to store the information (`shiftedLines` + `linesLength`), stored in a `Map<InputFileContext, DataStructure>`.
We then don't rely on the external code, to the price of some memory as we never clear out previous file data.
WDYT would be the best?

Reply:
I'm not sure the first method is future proof, as we will do some x-file analysis soon, as helm requires it.
I believe we're somehow bound to the second method.
But can't we just calculate the shiftedLines and lineLength on demand when we want to raise something there?
I think we don't need to save the data because we can always recalculate it based on inputfile.contents

Reply:
I will move forward the `Map<...>` solution then, seems more bulletproof.
Computing the `shiftedLines` "on the fly" could be possible but would require modifications in lower API code (we don't have access to the tree node when shifting issues + we would need to find the redirect comment), please see [my response to Marcin about it](https://github.com/SonarSource/sonar-iac/pull/968#issuecomment-1833501874) (and feel free to provide your input about it!).